### PR TITLE
345 mongo permission

### LINKED ISSUE #345: mongo does not have write permission in eks
## To Reproduce

- Follow https://notes.deciphernow.com/t/deploy-greymatter-to-amazon-eks/770 to setup a greymatter cluster in eks off of the release-2.0 branch (tested with `355140876b195386654ba1063e4a5ec8ebdfd853`)

or if you already have one deployed, change your kubectl with
```bash
aws eks --region us-east-1 update-kubeconfig --name prod1
```
- see that both the internal-mongo and external mongo pods throw the error 
```
2019-10-22T18:04:47.767+0000 I CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
about to fork child process, waiting until server is ready for connections.
forked process: 17
2019-10-22T18:04:47.768+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2019-10-22T18:04:47.854+0000 I CONTROL  [initandlisten] MongoDB starting : pid=17 port=27017 dbpath=/data/db 64-bit host=internal-data-mongo-0
2019-10-22T18:04:47.854+0000 I CONTROL  [initandlisten] db version v4.0.3
2019-10-22T18:04:47.854+0000 I CONTROL  [initandlisten] git version: 7ea530946fa7880364d88c8d8b6026bbc9ffa48c
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.0.2g  1 Mar 2016
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] allocator: tcmalloc
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] modules: none
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] build environment:
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten]     distmod: ubuntu1604
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten]     distarch: x86_64
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten]     target_arch: x86_64
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] options: { net: { bindIp: "127.0.0.1", port: 27017, ssl: { mode: "disabled" } }, processManagement: { fork: true, pidFilePath: "/tmp/docker-entrypoint-temp-mongod.pid" }, systemLog: { destination: "file", logAppend: true, path: "/proc/1/fd/1" } }
2019-10-22T18:04:47.855+0000 I STORAGE  [initandlisten] exception in initAndListen: IllegalOperation: Attempted to create a lock file on a read-only directory: /data/db, terminating
2019-10-22T18:04:47.855+0000 I NETWORK  [initandlisten] shutdown: going to close listening sockets...
2019-10-22T18:04:47.855+0000 I NETWORK  [initandlisten] removing socket file: /tmp/mongodb-27017.sock
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] now exiting
2019-10-22T18:04:47.855+0000 I CONTROL  [initandlisten] shutting down with code:100
ERROR: child process failed, exited with error number 100
To see additional information in this output, start without the "--fork" option.
```

### PR DESCRIPTION
**Pull request by dgoldstein1**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

closes #345 

2. What **changes to custom.yaml** are required?

- before starting, change `environment` to `eks`

3. Have you documented any additional setup steps or configurations options?

- none needed

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

- Followed https://notes.deciphernow.com/t/deploy-greymatter-to-amazon-eks/770 to create a new eks cluster
- change `greymatter.yaml` to use `eks` and `greymatter-secrets.yaml` with username / password
- modified `greymatter/requirements.yaml` to use local versions of `data`
- installed normally
```
david@david-Oryx-Pro:~/dev/work/helm-charts$ helm install greymatter -f greymatter.yaml -f greymatter-secrets.yaml --name gm
...
NOTES:
Grey Matter 2.0.4 has been installed.

gm deployed to namespace "default" at 04:15:44 on 10/24/04


NOTE: It may take a few minutes for the installation to become stable.
    You can watch the status of the pods by running 'kubectl get pods -w -n default'

```
- went into mongo pod and made sure there is data, inserted a new entry to track
```
david@david-Oryx-Pro:~/dev/work/helm-charts$ kubectl get pods | grep mongo
data-mongo-0                             1/1     Running            0          81s
internal-data-mongo-0                    1/1     Running            0          81s
david@david-Oryx-Pro:~/dev/work/helm-charts$ kubectl exec -it data-mongo-0 /bin/sh
$ mongo -u mongo -p mongo 127.0.0.1:27017/gmdata
MongoDB shell version v4.0.3
connecting to: mongodb://127.0.0.1:27017/gmdata
Implicit session: session { "id" : UUID("fbf2cc47-5471-4f33-a67a-72dd6612e934") }
MongoDB server version: 4.0.3
Welcome to the MongoDB shell.
For interactive help, type "help".
For more comprehensive documentation, see
	http://docs.mongodb.org/
Questions? Try the support group
	http://groups.google.com/group/mongodb-user
2019-10-24T16:17:30.223+0000 I STORAGE  [main] In File::open(), ::open for '//.mongorc.js' failed with Unknown error
Server has startup warnings: 
2019-10-24T16:16:04.702+0000 I STORAGE  [initandlisten] 
2019-10-24T16:16:04.702+0000 I STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
2019-10-24T16:16:04.702+0000 I STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
2019-10-24T16:16:10.215+0000 W CONTROL  [initandlisten] 
2019-10-24T16:16:10.215+0000 W CONTROL  [initandlisten] 
2019-10-24T16:16:10.215+0000 I CONTROL  [initandlisten] 
---
Enable MongoDB's free cloud-based monitoring service, which will then receive and display
metrics about your deployment (disk utilization, CPU, operation statistics, etc).

The monitoring data will be available on a MongoDB website with a unique URL accessible to you
and anyone you share the URL with. MongoDB may use this information to make product
improvements and to suggest MongoDB products and deployment options to you.

To enable free monitoring, run the following command: db.enableFreeMonitoring()
To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
---

> db.oids.count()
2
> db.oids.insert({"working?":true})
WriteResult({ "nInserted" : 1 })
```
- delete pod and see that data persists
```
david@david-Oryx-Pro:~/dev/work/helm-charts$ kubectl delete pod data-mongo-0 
pod "data-mongo-0" deleted
david@david-Oryx-Pro:~/dev/work/helm-charts$ kubectl exec -it data-mongo-0 /bin/sh
$ mongo -u mongo -p mongo 127.0.0.1:27017/gmdata
MongoDB shell version v4.0.3
connecting to: mongodb://127.0.0.1:27017/gmdata
Implicit session: session { "id" : UUID("1bd6b88c-dcf8-4761-938f-b68c94a77ad2") }
MongoDB server version: 4.0.3
Welcome to the MongoDB shell.
For interactive help, type "help".
For more comprehensive documentation, see
	http://docs.mongodb.org/
Questions? Try the support group
	http://groups.google.com/group/mongodb-user
2019-10-24T16:18:53.703+0000 I STORAGE  [main] In File::open(), ::open for '//.mongorc.js' failed with Unknown error
Server has startup warnings: 
2019-10-24T16:18:40.612+0000 I STORAGE  [initandlisten] 
2019-10-24T16:18:40.612+0000 I STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
2019-10-24T16:18:40.612+0000 I STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
2019-10-24T16:18:45.457+0000 W CONTROL  [initandlisten] 
2019-10-24T16:18:45.457+0000 W CONTROL  [initandlisten] 
2019-10-24T16:18:45.457+0000 I CONTROL  [initandlisten] 
---
Enable MongoDB's free cloud-based monitoring service, which will then receive and display
metrics about your deployment (disk utilization, CPU, operation statistics, etc).

The monitoring data will be available on a MongoDB website with a unique URL accessible to you
and anyone you share the URL with. MongoDB may use this information to make product
improvements and to suggest MongoDB products and deployment options to you.

To enable free monitoring, run the following command: db.enableFreeMonitoring()
To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
---

> db.oids.find({"working?":true}).pretty()
{ "_id" : ObjectId("5db1cebbf1bfecfe63ba2d61"), "working?" : true }
```
- go to voyager-edge ingress endpoint as ELB load balancer, see dashboard 
Screenshot from 2019-10-24 12-23-23
URL: https://a88b1111ef67911e98b601666bc5d15c-631372704.us-east-1.elb.amazonaws.com

### CODE REVIEW

File: data/templates/mongo.yaml [hunk 1]
```diff
@@ -17,6 +17,12 @@ spec:
       labels:
         app: {{ .Values.mongo.name }}
     spec:
+      {{- if and .Values.global.environment (eq .Values.global.environment "eks") }}
+      securityContext:
+        runAsUser: 1099
```
Comment (chrisbsmith) [inline data/templates/mongo.yaml]: Is `1099` the default value for these in an EKS cluster?

File: data/templates/mongo.yaml [hunk 2]
```diff
@@ -62,8 +68,7 @@ spec:
   - metadata:
       name: data
     spec:
-      accessModes:
-        - ReadWriteOnce
+      accessModes: [ "ReadWriteOnce" ]
```
Comment (chrisbsmith) [inline data/templates/mongo.yaml]: Why did you have to make this change?

Comment (dgoldstein1) [reply data/templates/mongo.yaml]: didn't mean to, reverted this.

File: data/templates/mongo.yaml (hunk 1 above)
Comment (dgoldstein1) [reply data/templates/mongo.yaml]: The default mounting permissions for stateful sets in eks are `root:root`. I got `1099` from [online forms](https://stackoverflow.com/questions/43544370/kubernetes-how-to-set-volumemount-user-group-and-file-permissions). 

I experimented a little more with permissions and it seems to work with `2000`, which is the same as what we're using for prometheus and postgres, so I've changed it to that.

File: greymatter/requirements.yaml [hunk 3]
```diff
@@ -6,11 +6,13 @@ dependencies:
   - name: data
     version: '2.0.6'
```
[the remark this answers is not in the document]
Comment (dgoldstein1) [reply greymatter/requirements.yaml]: :heavy_check_mark:

**Pull request merged**